### PR TITLE
Implement user management page & refactor new user invitations

### PR DESCRIPTION
**Pull request by acrellin**
This PR introduces a user management page accessible by users with the "Manage users" ACL. From there, admin users can easily remove any user's group and/or stream access, and bulk invite new users.

Screen capture of managing users' group/stream access:

user_mgmt_delete_user_group_stream

Screen capture of bulk inviting new users:

bulk_invite_users_end_to_end

### CODE REVIEW

File: skyportal/handlers/api/group.py [hunk 1]
```diff
@@ -381,49 +379,37 @@ def post(self, group_id, *ignored_args):
...
+            # Ensure user has sufficient stream access to be added to group
+            if group.streams:
+                user_stream_ids = [
+                    su.stream_id
+                    for su in DBSession()
+                    .query(StreamUser)
+                    .filter(StreamUser.user_id == user.id)
+                    .all()
+                ]
+                if not all([stream.id in user_stream_ids for stream in group.streams]):
```
Comment (dmitryduev) [inline skyportal/handlers/api/group.py:398]: The culmination of months of thinking about ACLs 😂

File: skyportal/handlers/api/invitations.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        if data.get("userEmail") is None or data.get("userEmail") in [
+            "",
+            "None",
+            "null",
+        ]:
+            return self.error("Missing required parameter `userEmail`")
+        user_email = data["userEmail"].strip()
+
+        if not data.get("groupIDs") is not None:
```
Comment (dmitryduev) [inline skyportal/handlers/api/invitations.py:75]: is None?

File: static/js/components/UserManagement.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                    <Chip
+                      label={stream.name}
+                      onDelete={() => {
+                        handleClickRemoveUserStreamAccess(user.id, stream.id);
+                      }}
+                      key={stream.id}
+                      id={`deleteStreamUserButton_${user.id}_${stream.id}`}
+                    />
+                  ))}
+                </TableCell>
```
Comment (dmitryduev) [inline static/js/components/UserManagement.jsx:158]: Let's leave it for another PR, but would also be nice to be able to add stream access here as well.

File: static/js/components/SidebarAndHeader.jsx.template [hunk 4]
```diff
@@ -185,6 +187,23 @@ const SidebarAndHeader = () => {
           {%- endif -%}
 
           {%- endfor %}
+          {currentUser.acls?.includes("System admin") && (
```
Comment (dmitryduev) [inline static/js/components/SidebarAndHeader.jsx.template:190]: || currentUser.acls?.includes("Manage users")  ?

File: static/js/components/UserManagement.jsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+  if (!allUsers?.length || !currentUser?.username?.length) {
+    return (
+      <div>
+        <CircularProgress />
+      </div>
+    );
+  }
+
+  if (!currentUser.acls?.includes("System admin")) {
```
Comment (dmitryduev) [inline static/js/components/UserManagement.jsx:52]: Same here, or should also adjust the expected ACLs on the backend

**Review by dmitryduev — APPROVED** [review]
LGTM, just a couple minor comments!

File: skyportal/handlers/api/invitations.py (hunk 2 above)
Comment (acrellin) [reply skyportal/handlers/api/invitations.py:75]: Yes of course, thanks, this will be updated (changed multiple times and came out in bad shape 😂)

File: static/js/components/SidebarAndHeader.jsx.template (hunk 4 above)
Comment (acrellin) [reply static/js/components/SidebarAndHeader.jsx.template:190]: 👍 Will update

File: static/js/components/UserManagement.jsx (hunk 3 above)
Comment (acrellin) [reply static/js/components/UserManagement.jsx:158]: Yes, and also perhaps add existing users to groups

File: static/js/components/UserManagement.jsx (hunk 5 above)
Comment (acrellin) [reply static/js/components/UserManagement.jsx:52]: Will update

File: skyportal/handlers/api/group.py (hunk 1 above)
Comment (acrellin) [reply skyportal/handlers/api/group.py:398]: :+1: :joy:

File: skyportal/handlers/api/invitations.py (hunk 2 above)
Comment (acrellin) [reply skyportal/handlers/api/invitations.py:75]: Done

File: static/js/components/UserManagement.jsx (hunk 3 above)
Comment (acrellin) [reply static/js/components/UserManagement.jsx:158]: Will capture this in an issue/story

File: static/js/components/SidebarAndHeader.jsx.template (hunk 4 above)
Comment (acrellin) [reply static/js/components/SidebarAndHeader.jsx.template:190]: Done

**Comment by acrellin** [comment]
@dmitryduev thanks for the review and good suggestions -- I've implemented them all. Look okay?

File: static/js/components/UserManagement.jsx (hunk 5 above)
Comment (acrellin) [reply static/js/components/UserManagement.jsx]: Done

File: skyportal/handlers/api/invitations.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                "Invalid value provided for `groupIDs`; unable to parse "
+                "all list items to integers."
+            )
+        groups = DBSession().query(Group).filter(Group.id.in_(group_ids)).all()
+
+        if data.get("streamIDs") is not None and data.get("streamIDs") not in [
+            "",
+            "null",
+            "None",
+        ]:
```
Comment (profjsb) [inline skyportal/handlers/api/invitations.py:90]:
```suggestion
        if data.get("streamIDs") not in [None, "", "null", "None"]:
```

**Review by profjsb — APPROVED** [review]
Really nice, @acrellin. Small suggested change then good to go.
